runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -102,14 +97,24 @@ public ILogData load(Long aLong) throws Exception {
      * @param stream         The streams which will belong on this entry.
      * @param data           The data to write.
      * @param backpointerMap
+     * @param epoch          To verify that the token is acquired in the same epoch as the write.
      */
-    public void write(long address, Set<UUID> stream, Object data, Map<UUID, Long> backpointerMap,
-                      Map<UUID, Long> streamAddresses)
+    public void epochedWrite(long address, Set<UUID> stream, Object data, Map<UUID, Long> backpointerMap,
+                             Map<UUID, Long> streamAddresses, long epoch)
             throws OverwriteException {
-        write(address, stream, data, backpointerMap, streamAddresses, null);
+        epochedWrite(address, stream, data, backpointerMap, streamAddresses, null, epoch);
     }
 
-    public void write(long address, Set<UUID> stream, Object data, Map<UUID, Long> backpointerMap,
+    public void epochedWrite(long address, Set<UUID> stream, Object data, Map<UUID, Long> backpointerMap,
```

Comment:
my larger comment is that why did we not choose to have epoch as part of the sequencer token ?